core/src/test/scala/org/apache/spark/deploy/history/FsHistoryProviderSuite.scala

```diff
@@ -1124,15 +1124,13 @@ class FsHistoryProviderSuite extends SparkFunSuite with Matchers with Logging {
       SparkListenerApplicationEnd(5L))
     val mockedFs = spy(provider.fs)
     doThrow(new AccessControlException("Cannot read accessDenied file")).when(mockedFs).open(
-      argThat((path: Path) => path.getName.toLowerCase(Locale.ROOT) == "accessdenied"))
+      argThat((path: Path) => path.getName.toLowerCase(Locale.ROOT) == "accessdenied" &&
+        clock.getTimeMillis() < 1533132471 + 24 * 60 * 60 * 1000))
```

Comment:
nit: it would be great to have a more meaningful condition here. Maybe we could have a `var canReadAccessDenied: Boolean = false`, which is turned to `true` where there was the write file, in order to simulate the mutation in policies for the file.

Reply:
Yep. No problem.